langchain4j-mcp/src/test/java/dev/langchain4j/mcp/client/transport/http/McpHttpTransportTest.java

```diff
@@ -0,0 +1,123 @@
+package dev.langchain4j.mcp.client.transport.http;
+
+import static dev.langchain4j.model.openai.OpenAiChatModelName.GPT_4_O_MINI;
+
+import dev.langchain4j.agent.tool.ToolExecutionRequest;
+import dev.langchain4j.agent.tool.ToolSpecification;
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.data.message.ToolExecutionResultMessage;
+import dev.langchain4j.data.message.UserMessage;
+import dev.langchain4j.mcp.McpToolProvider;
+import dev.langchain4j.mcp.client.DefaultMcpClient;
+import dev.langchain4j.mcp.client.McpClient;
+import dev.langchain4j.mcp.client.transport.McpTransport;
+import dev.langchain4j.model.openai.OpenAiChatModel;
+import dev.langchain4j.model.output.Response;
+import dev.langchain4j.service.tool.ToolExecutor;
+import dev.langchain4j.service.tool.ToolProvider;
+import dev.langchain4j.service.tool.ToolProviderResult;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.MountableFile;
+
+@EnabledIfEnvironmentVariable(named = "OPENAI_API_KEY", matches = ".+")
+public class McpHttpTransportTest {
+
+    GenericContainer<?> container = new GenericContainer<>("docker.io/node:23")
+            .withCopyFileToContainer(MountableFile.forClasspathResource("everything"), "/everything")
+            .withLogConsumer(outputFrame -> System.out.println(outputFrame.getUtf8String()))
+            .withWorkingDirectory("/everything")
+            .withCommand("sh", "-c", "npm install && npm run build && node dist/sse.js")
+            .withExposedPorts(3001)
+            .waitingFor(Wait.forHttp("/").forStatusCode(404));
+
+    @Test
+    public void test() throws Exception {
+        container.start();
+        Integer port = container.getMappedPort(3001);
+        McpClient mcpClient = null;
+        try {
+            OpenAiChatModel chat = OpenAiChatModel.builder()
```

Comment:
Would ChatModelMock work for the test as well?

Reply:
I rewrote it to not use a chat model at all